install_cluster.py

```diff
@@ -0,0 +1,260 @@
+        _create_config_file(config_dict, node.name)
+
+
+def _install_instances(instances_dict, rpm_download_link):
+    for instance_type in INSTANCES_TYPES:
+        logger.info('installing %s instances', instance_type)
+        for instance in instances_dict[instance_type]:
+            logger.info('Copying the %s directory to %s',
+                        LOCAL_INSTALL_CLUSTER_DIR, instance.name)
+            instance.put_dir(LOCAL_INSTALL_CLUSTER_DIR,
+                             REMOTE_INSTALL_CLUSTER_DIR,
+                             override=True)
+            logger.info('Installing Cloudify RPM on %s', instance.name)
+            instance.run_command('curl -o {0} {1}'.format(REMOTE_RPM_PATH,
+                                                          rpm_download_link))
+            instance.run_command(
+                'yum install -y {}'.format(REMOTE_RPM_PATH), use_sudo=True)
+
+            config_path = join(REMOTE_CONFIG_DIR,
+                               '{}_config.yaml'.format(instance.name))
+            instance.run_command(
+                'cp {0} /etc/cloudify/config.yaml'.format(config_path))
+
+            logger.info('Installing %s', instance.name)
+            instance.run_command(
+                'cfy_manager install --private-ip {0} --public-ip {1}'.format(
+                    instance.private_ip, instance.public_ip))
+
+
+def _sort_instances_dict(instances_dict):
+    for _, instance_items in instances_dict.items():
+        if len(instance_items) > 1:
+            instance_items.sort(key=lambda x: int(x.name.rsplit('-', 1)[1]))
+
+
+def _get_instances_dict(config):
+    instances_dict = {instance_type: [] for instance_type in INSTANCES_TYPES}
+    username = config.get('machine_username')
+    for node_name, node_dict in config.get('existing_vms').items():
+        new_vm = CfyNode(node_dict.get('private_ip'),
+                         node_dict.get('public_ip'),
+                         JUMP_HOST_SSH_KEY_PATH,
+                         username,
+                         node_name)
+        instances_dict[new_vm.node_type].append(new_vm)
+    _sort_instances_dict(instances_dict)
+    return instances_dict
+
+
+def _create_install_cluster_directory():
+    logger.info('Creating `remote_cluster_install` directory')
+    run(['rm', '-rf', '{}_old'.format(LOCAL_INSTALL_CLUSTER_DIR)])
+    if exists(LOCAL_INSTALL_CLUSTER_DIR):
+        run(['mv', LOCAL_INSTALL_CLUSTER_DIR,
+             '{0}_old'.format(LOCAL_INSTALL_CLUSTER_DIR)])
+    os.mkdir(LOCAL_INSTALL_CLUSTER_DIR)
+    copy(JUMP_HOST_LICENSE_PATH,
+         join(LOCAL_INSTALL_CLUSTER_DIR, 'license.yaml'))
+
+
+def _show_manager_ips(manager_nodes):
+    managers_str = ''
+    for manager in manager_nodes:
+        managers_str += '{0}: {1}\n'.format(manager.name, manager.public_ip)
+    logger.info('In order to connect to one of the managers, use one of the '
+                'following IPs:\n%s', managers_str)
```

Comment:
If this output is for ease of ssh connection, it might be worth putting the hostname there too.

Reply:
Right, I'll change the output to be the whole connection string.